Fix an issue that data in Cookies may be lost

rubyu opened the pull request:
I encountered a problem that a data saved to Cookies by JavaScript was sometimes to be lost only in the last one. I tried and found out that [CookieManager](http://developer.android.com/reference/android/webkit/CookieManager.html) managed Cookies asynchronously and because of that the changes applied to Cookies after the last synchronization process was lost. Therefore, it is necessary to specify the timings when it to sync Cookies manifestly. I guess calling [`CookieManager.flush()`](http://developer.android.com/reference/android/webkit/CookieManager.html#flush()) in [`onPause()`](https://github.com/ankidroid/Anki-Android/blob/develop/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java#L905) and [`onDestroy()`](https://github.com/ankidroid/Anki-Android/blob/develop/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java#L933)  in [AbstractFlashcardViewer.java](https://github.com/ankidroid/Anki-Android/blob/develop/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java) will be sufficient.

And I also found a little bit of a problem that [WebView](http://developer.android.com/reference/android/webkit/WebView.html) instances were not released on [`onDestroy()`](https://github.com/ankidroid/Anki-Android/blob/develop/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java#L933) in [AbstractFlashcardViewer.java](https://github.com/ankidroid/Anki-Android/blob/develop/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java). [`WebView.destroy()`](http://developer.android.com/reference/android/webkit/WebView.html#destroy()) should be called after use as described in the document. This may cause a memory leak, though the effects due to it were highly unlikely.

hssm commented:
`onPause()` will be called before `onDestroy()` so there is no need to put it in both places. 

The rest looks fine, and thanks for the potential memory leak fix. Does this solution mean that the problem persists in API < 21?

rubyu commented:
I agree with you. I misunderstood. I will fix soon.

This problem happens on all versions of Android, and this can be solved by using `CookieManager.flush()`, added in API 21, in API < 21. Regarding API > 21, we should use `CookieSyncManager.sync()`, deprecated since API 21,  to solve this problem. Should I add an implementation for API > 21?

hssm commented:
You've got your `<` and `>` the wrong way around, but yes, please do that. In the CompatV10 class where you added a comment, replace it with the deprecated call.

rubyu commented:
Thank you for pointing that out! I learned A < N means A which is smaller than N.

I have deleted a duplicated call for `CookieManager.flush()` and added an implementation to support API < 21.

hssm commented:
Thanks. Just out of curiosity, what do you use cookies for in Anki? Does it work on the desktop client?

rubyu commented:
I'm learning English using a Anki deck created from COCA60K word list and two English-Japanese Epwing dictionary in recent years. The deck was good enough for beginners, but above a certain level it became cool toward me. It sometimes spat out a line or a single word tautologically. So, I tried to create new deck consisted of three English-Japanese dictionary, one English-English dictionary, one English-Japanese corpus and more. This new deck looked perfect but had a fatal problem -- too fat. 
I investigated the specification of Anki and found a possible solution. It was iframe. 
This is why I need cookies. Spongy deck + heavy HTML + JavaScript is the solution. This is not violating [Anki's 100MB limit](https://anki.tenderapp.com/kb/anki-ecosystem/are-there-limits-on-file-sizes-on-ankiweb) and can do anything with Javascript.

I tested this iframe+js+cookie deck on Anki(Windows) and Ankidroid. In both cases, this will work fine.

hssm commented:
Okay thanks for the explanation.
